R/tmb-methods.R

```diff
@@ -429,3 +429,152 @@ h_residuals_response <- function(object) {
   assert_class(object, "mmrm_tmb")
   component(object, "y_vector") - unname(fitted(object))
 }
+
+
+#' @describeIn mmrm_tmb_methods Simulates from an mmrm model object
+#' @importFrom stats simulate
+#' @exportS3Method
+#'
+#' @param object an object of class 'mmrm'
+#' @param nsim number of response vectors to simulate. Defaults to 1.
+#' @param seed an object specifying if and how the random number generator should be initialized (‘seeded’).
+#' @param newdata data frame for which to evaluate predictions.
+#' @param ...
+#'
+#' @return A \code{data.frame} of dimension [n, m] where n is the number of observations
+#' fitted in the model, and m is the number \code{nsim} of simulated responses.
+simulate.mmrm_tmb <- function(object, nsim = 1,
+                              newdata = NULL,
+                              method = "conditional",
+                              ...){
+
+  method <- match.arg(method)
+
+  if(is.null(newdata)){
+    # build data.frame of prediction vectors
+    ret <- fitted(object)
```

Comment:
in this way we ignore the missingness in the data. This is inconsistent with the new data approach